### PR TITLE
[GPU] Fix PR18171 regression

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - Use pShape() when allow_new_shape_infer=1 and get_tensor() for allow_new_shape_infer=0
 - Remove unnecessary condition about rank>4 in gemm-eltwise fusing 

### Tickets:
 -

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp [hunk 1]
```diff
@@ -1063,11 +1063,6 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
                 auto eltw_in_size = peer_node->get_output_layout();
                 if (eltw_in_size.is_dynamic())
                     return;
-                // When input rank > 4, fused eltwise to gemm should be converted to 4 dim in init_onednn_primitive_attribute()
-                // But current init_onednn_primitive_attribute() cannot handle dynamic shape case.
-                auto eltw_in_rank = fused_node->get_output_layout().get_rank();
-                if ((fused_node->is_type<gemm>()) && (eltw_in_rank > 4))
-                    return;
             }
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:1066]: Isn't this needed any more for the original purpose too?

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp:1066]: No. My approach about this case is little bit far from existing unit test of gemm-eltwise fusing with dynamic shape.
So actual gemm-eltwise fusing case doesn't make get_tensor() issue in add_onednn_optimization_attributes pass.

**Review by yeonbok — APPROVED** [review]

**Comment by yeonbok** [comment]
@isanghao  @byungilm Please review this

**Review by isanghao — APPROVED** [review]
Looks good to me and it resolves unittest and accuracy issue.

**Pull request merged**